virtool/tasks/runner.py

```diff
@@ -19,24 +19,42 @@ def __init__(
         self._data = data
         self._tasks_client = tasks_client
         self.app = app
+        self.current_task = None
 
     async def run(self):
-        logging.info("Started task runner")
+        task_id = None
 
         try:
             while True:
                 logging.info("Waiting for next task")
-
                 task_id = await self._tasks_client.pop()
 
                 await self.run_task(task_id)
 
+                await asyncio.shield(self.current_task)
+
                 logging.info("Finished task: %s", task_id)
 
         except asyncio.CancelledError:
-            logging.info("Stopped task runner")
+            if task_id is not None:
+                logging.info("Recieved stop signal; awaiting task completion")
+
+                try:
+                    await asyncio.wait_for(asyncio.shield(self.current_task), 600)
+
+                    logging.info("Finished task: %s", task_id)
+
+                except asyncio.TimeoutError:
+                    logging.info("Task %s timed out", task_id)
```

Comment:
This should never happen and is considered an error state. Better wording may be _Timed out while waiting for task to complete_.

```suggestion
                except asyncio.TimeoutError:
                    logging.critical("Task %s timed out", task_id)
```